sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/Filter.cs

```diff
@@ -0,0 +1,719 @@
+                    }
+                    else if (fieldType == typeof(Uri))
+                    {
+                        Expression toStringCall = Expression.Call(fieldExpression, UriToStringMethodInfo);
+
+                        Expression fieldValueOrEmptyString = Expression.Condition(Expression.Equal(fieldExpression, Expression.Constant(null)), Expression.Constant(string.Empty), toStringCall);
+
+                        Expression indexOfCall = Expression.Call(fieldValueOrEmptyString, StringIndexOfMethodInfo, Expression.Constant(this.comparand), Expression.Constant(StringComparison.OrdinalIgnoreCase));
+
+                        switch (this.predicate)
+                        {
+                            case Predicate.Equal:
+                                // (fieldValue?.ToString() ?? string.Empty).Equals(this.comparand, StringComparison.OrdinalIgnoreCase)
+                                return Expression.Call(fieldValueOrEmptyString, StringEqualsMethodInfo, Expression.Constant(this.comparand), Expression.Constant(StringComparison.OrdinalIgnoreCase));
+                            case Predicate.NotEqual:
+                                // !(fieldValue?.ToString() ?? string.Empty).Equals(this.comparand, StringComparison.OrdinalIgnoreCase)
+                                return Expression.Not(Expression.Call(fieldValueOrEmptyString, StringEqualsMethodInfo, Expression.Constant(this.comparand), Expression.Constant(StringComparison.OrdinalIgnoreCase)));
+                            case Predicate.Contains:
+                                // fieldValue => (fieldValue?.ToString() ?? string.Empty).IndexOf(this.comparand, StringComparison.OrdinalIgnoreCase) != -1;
+                                return Expression.NotEqual(indexOfCall, Expression.Constant(-1));
+                            case Predicate.DoesNotContain:
+                                // fieldValue => (fieldValue?.ToString() ?? string.Empty).IndexOf(this.comparand, StringComparison.OrdinalIgnoreCase) == -1;
+                                return Expression.Equal(indexOfCall, Expression.Constant(-1));
+                            default:
+                                this.ThrowOnInvalidFilter(fieldType);
+                                break;
+                        }
+                    }
+                    else if ((nullableUnderlyingType = Nullable.GetUnderlyingType(fieldType)) != null)
+                    {
+                        // make a recursive call for the underlying type
+                        return ProduceComparatorExpressionForSingleFieldCondition(fieldExpression, nullableUnderlyingType, true);
+                    }
+                    else
+                    {
+                        this.ThrowOnInvalidFilter(fieldType);
+                    }
+
+                    break;
+            }
+
+            // TODO: Throwing an exception for now instead of returning null. This is an unknown state and should be investigated.
```

Comment:
Very difficult to find this in the code, if there are many TODO recommend documenting in a different file or add on the top of the file to catch it later.

Reply:
A lot of them are still in progress. I'll compile a list of TODOs later.